src/components/Container.jsx

```diff
@@ -27,20 +27,20 @@ export class Container extends addMoreDecorator(Component) {
     this.onControlRemove = this.onControlRemove.bind(this);
     this.onEventTrigger = this.onEventTrigger.bind(this);
     this.showNotification = this.showNotification.bind(this);
-  }
 
-  componentWillMount() {
     const initScript = this.props.metadata.events && this.props.metadata.events.onFormInit;
     let updatedTree;
     if (initScript) {
-      updatedTree = new ScriptRunner(this.state.data, this.props.patient).execute(initScript);
-      this.setState({ data: updatedTree });
+      try {
+        updatedTree = new ScriptRunner(this.state.data, this.props.patient).execute(initScript);
+        updatedTree = executeEventsFromCurrentRecord(updatedTree, updatedTree, this.props.patient);
```

Comment:
@angshu the `executeEventsFromCurrentRecord` function seems to be moved inside the if block so that it gets executed only when there is a onFormInit event. Is this intentional? Previously it seems to be invoked outside of if.

Reply:
yeah, it should be outside. will fix